src/actions/create-async-action.ts

```diff
@@ -17,14 +17,25 @@ export default function createAsyncAction<
       meta: { phase: Phase.Optimistic },
     };
 
-    const payload = await effect(input);
+    try {
+      const payload = await effect(input);
 
-    yield {
-      type: actionType,
-      payload,
-    };
+      yield {
+        type: actionType,
+        payload,
+      };
+
+      return payload;
+    } catch (error) {
+      yield {
+        type: actionType,
+        error: true,
+        payload: error,
+      };
 
-    return payload;
+      // TODO: Determine how to treat known errors.
```

Comment:
A couple of ideas:
- Return a `Promise.reject(...)` with an orphaned noop catch branch.
- Change the return signature to be `ActionSuccess<T> | ActionFailure<unknown>`.
- Nothing.

It bothers me that known errors still result in a rejected dispatch, but that's a problem for another PR.